src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function asas(arg) {
+    if (arg !== undefined) {
+      sum += arg;
+
+      return (a) => asas(a);
+    } else {
+      const tmp = sum;
+
+      sum = 0;
+
+      return tmp;
+    }
```

Comment:
```suggestion
    } 
    
    const tmp = sum;

    sum = 0;

    return tmp;
  
```
try to avoid else where it is possible